userspace/libscap/engine/noop/noop.h

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#pragma once
+
+#include "engine_handle.h"
+#include "scap_vtable.h"
+
+typedef struct scap scap_t;
+typedef struct ppm_evt_hdr scap_evt;
+typedef struct scap_stats scap_stats;
+
+struct noop_engine* noop_alloc_handle(scap_t* main_handle, char* lasterr_ptr);
+void noop_free_handle(struct scap_engine_handle engine);
+int noop_close_engine(struct scap_engine_handle engine);
+int32_t noop_next(struct scap_engine_handle handle, scap_evt** pevent, uint16_t* pcpuid);
+int32_t noop_start_capture(struct scap_engine_handle engine);
+int32_t noop_stop_capture(struct scap_engine_handle engine);
+int32_t unimplemented_op(char* err, size_t err_size);
+int32_t noop_configure(struct scap_engine_handle engine, enum scap_setting setting, unsigned long arg1, unsigned long arg2);
+int32_t noop_get_stats(struct scap_engine_handle engine, scap_stats* stats);
+int32_t noop_get_n_tracepoint_hit(struct scap_engine_handle engine, long* ret);
+uint32_t noop_get_n_devs(struct scap_engine_handle engine);
+uint64_t noop_get_max_buf_used(struct scap_engine_handle engine);
+
+const struct scap_vtable scap_noop_engine;
```

Comment:
Building the simple `scap-open` example over this PR, I obtain this:

```
../../engine/source_plugin/libscap_engine_source_plugin.a(source_plugin.c.o):/home/andrea/git_projects/libs/userspace/libscap/engine/source_plugin/../noop/noop.h:40: definizione multipla di "scap_noop_engine"
../../engine/nodriver/libscap_engine_nodriver.a(nodriver.c.o):/home/andrea/git_projects/libs/userspace/libscap/engine/nodriver/../noop/noop.h:40: definito qui per la prima volta
../../engine/noop/libscap_engine_noop.a(noop.c.o):/home/andrea/git_projects/libs/userspace/libscap/engine/noop/noop.h:40: definizione multipla di "scap_noop_engine"
../../engine/nodriver/libscap_engine_nodriver.a(nodriver.c.o):/home/andrea/git_projects/libs/userspace/libscap/engine/nodriver/../noop/noop.h:40: definito qui per la prima volta
collect2: error: ld returned 1 exit status
libscap/examples/01-open/CMakeFiles/scap-open.dir/build.make:102: recipe for target 'libscap/examples/01-open/scap-open' failed
make[3]: *** [libscap/examples/01-open/scap-open] Error 1
CMakeFiles/Makefile2:634: recipe for target 'libscap/examples/01-open/CMakeFiles/scap-open.dir/all' failed
make[2]: *** [libscap/examples/01-open/CMakeFiles/scap-open.dir/all] Error 2
CMakeFiles/Makefile2:641: recipe for target 'libscap/examples/01-open/CMakeFiles/scap-open.dir/rule' failed
make[1]: *** [libscap/examples/01-open/CMakeFiles/scap-open.dir/rule] Error 2
Makefile:325: recipe for target 'scap-open' failed
make: *** [scap-open] Error 2
```
Why do we need to define here the `scap_noop_engine` ? Maybe we could remove it, since it seems to be not necessary. WDYT @gnosek ?

Reply:
I got the same issue (also when compiling it in Falco)